jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -124,4 +127,24 @@ private void addJsonbTransientSpecificDiagnostics(Diagnostic d) {
         d.setMessage(JsonbConstants.ERROR_MESSAGE_JSONB_TRANSIENT);
         d.setCode(JsonbConstants.DIAGNOSTIC_CODE_ANNOTATION_TRANSIENT_FIELD);
     }
+    
+    private boolean hasJsonbTransient(IAnnotatable annotable) throws JavaModelException {
+        for (IAnnotation annotation : annotable.getAnnotations()) {
+            String annotationName = annotation.getElementName();
+            if (annotationName.equals(JsonbConstants.JSONB_TRANSIENT))
+                return true;
+        }
+        return false;
+    }
+    
+    private boolean hasNonJsonbTransientAnnotation(IAnnotatable annotable) throws JavaModelException {
+        for (IAnnotation annotation : annotable.getAnnotations()) {
+            String annotationName = annotation.getElementName();
+            if (annotationName.startsWith(JsonbConstants.JSONB_PREFIX) 
```

Comment:
Can we be sure that all other annotations starting with "Jsonb" are considered "other Json Binding annotations"? Thinking about a case where a developer creates a custom annotation with the same prefix (though seems unlikely). I guess the alternative would be to create constants for all the known jsonb annotations.  

What do you think @jooyoojoo? I am fine to leave it as is

Reply:
Yes this is a good point. I made a list of all known Jsonb annotations and check if it contains the annotation name instead. In the constant class, I assigned each individual Jsonb annotation to a variable so that they can be reused for other features in the future.